gpii/node_modules/lifecycleManager/src/LifecycleManager.js

```diff
@@ -490,33 +517,82 @@ var gpii = fluid.registerNamespace("gpii");
     };
 
     /**
+     * The lifecycleManager queue is used to hold the high-level actions that needs to happen,
+     * such as starting the login process, starting logout process, starting the update process.
+     * Each item in the queue should have the format: { func: <functionToCall>, args: <Arguments> }
+     * Where <functionToCall> is a function that returns a promise for when it's complete (including side-effects)
+     * and the arguments are the arguments that should be passed to the function.
+     * The queue is run sequentially, and an item is considered "done" once the promise returned by its
+     * function is resolved.
+     */
+    gpii.lifecycleManager.addToQueue = function (that, queue, item) {
+        item.promise = fluid.promise();
+        queue.push(item);
+        if (!that.isProcessingQueue) {
+            that.processQueue();
+        }
+        return item.promise;
+    };
+
+    gpii.lifecycleManager.clearQueue = function (queue) {
+        while (queue.length > 0) {
+            var item = queue.shift();
+            item.promise.reject("clearQueue: clearing lifecycleManager promise from queue - remaining items: " + queue.length);
+        }
+    };
+
+    gpii.lifecycleManager.processQueue = function (that, queue) {
+        // mark that we're currently processing queue
+        that.isProcessingQueue = true;
+        // pick first item and process
+        var item = queue.shift();
+        var promise = item.func(item.args);
+
+        promise.then(function (val) {
+            // resolve the original promise of the item;
+            item.promise.resolve(val);
+            // process next item in queue if it exists
+            if (queue.length > 0) {
+                that.processQueue(queue);
+            } else {
+                that.isProcessingQueue = false;
+            }
+        }, function (error) {
+            fluid.log(fluid.logLevel.ERROR, "An error occurred in an item of the lifecyclemanagers queue" +
+                "(" + item.func + "), so clearing queue. Error was", error);
```

Comment:
Seeing the comment above, encoding the function in some string form will make this error report rather clearer